### DISCUSSION TITLE
Confusion on `show_completions_on_input` vs `show_inline_completions`

### DISCUSSION THREAD

**Discussion by xpe** [zed-industries/zed#22371] [Feature Requests]
These two config settings are confusingly similar:
- `show_completions_on_input`
- `show_inline_completions`

The answers to the following questions should be obvious from reading the documentation...

- How do these two settings interact?
- Which is "dominant"?

... but they are not.

The core problem stems from their confusing names. Both have `show_` and have to do with `completions`. So what is the difference between `on_input` and `inline`? Please clarify. 

Even better, please rename one or both, since that is probably the only way to settle this confusion once and for all. If someone has to go to the documentation page just to get unconfused, that is a hint that current design choices are not working well enough.

Ideas to consider:

- Rename `show_inline_completions ` to `auto_show_completion_ghost_text`?
- Rename `show_completions_on_input` to `auto_trigger_completion_menu`?

## Current Documentation

From [`zed/docs/src/configuring-zed.md`](https://github.com/zed-industries/zed/blob/main/docs/src/configuring-zed.md):

### [Show Inline Completions](https://github.com/zed-industries/zed/blob/main/docs/src/configuring-zed.md#show-inline-completions)

- Description: Whether to show inline completions as you type or manually by triggering `editor::ShowInlineCompletion`.
- Setting: `show_inline_completions`
- Default: `true`
- Options: boolean values

### [Show Completions On Input](https://github.com/zed-industries/zed/blob/main/docs/src/configuring-zed.md#show-completions-on-input)

- Description: Whether or not to show completions as you type.
- Setting: `show_completions_on_input`
- Default: `true`
- Options: boolean values

## Current Source Code

[`crates/language/src/language_settings.rs#L112`](https://github.com/zed-industries/zed/blob/main/crates/language/src/language_settings.rs#L112)

```rust
/// Controls whether inline completions
/// are shown immediately (true)
/// or manually by triggering
/// `editor::ShowInlineCompletion` (false).
pub show_inline_completions: bool,
```

[`crates/language/src/language_settings.rs#L141`](https://github.com/zed-industries/zed/blob/main/crates/language/src/language_settings.rs#L141)

```rust
/// Whether to pop the completions menu
/// while typing in an editor without
/// explicitly requesting it.
pub show_completions_on_input: bool,
```

**Comment by xpe** [comment]
To summarize two key take-aways: I recommend making two renames:

- rename `show_inline_completions ` to `auto_show_completion_ghost_text`
- rename `show_completions_on_input` to `auto_trigger_completion_menu`

**Reply by xpe** [reply]
I created [feature request #22372](https://github.com/zed-industries/zed/issues/22372) for this recommendation.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
